src/components/MapboxStaticImage.tsx

```diff
@@ -0,0 +1,81 @@
+import { useMemo, useRef, useState } from 'react';
+import { MapboxOptions, LngLatLike } from 'mapbox-gl';
+import useLayoutEffect from 'use-isomorphic-layout-effect';
+
+/**
+ * Props for {@link MapboxStaticImage}
+ */
+export interface MapboxStaticImageProps {
+  /** Mapbox access token. */
+  mapboxAccessToken: string,
+  /** Map customization for Mapbox static image. */
+  mapboxOptions: Pick<MapboxOptions, 'center' | 'zoom' | 'style'>
+}
+
+/**
+ * Renders a static map image using Mapbox Static Image API
+ *
+ * @internal
+ */
+export function MapboxStaticImage({
+  mapboxAccessToken,
+  mapboxOptions
+}: MapboxStaticImageProps): JSX.Element | null {
+  const mapContainer = useRef<HTMLDivElement>(null);
+  const [divDimension, setDivDimension] = useState<{ width: number, height: number }>();
+  useLayoutEffect(() => {
+    if (mapContainer.current) {
+      setDivDimension({
+        width: mapContainer.current.clientWidth,
+        height: mapContainer.current.clientHeight
+      });
+    }
+  }, []);
+
+  const staticMapboxStyle: React.HTMLAttributes<HTMLDivElement>['style'] = useMemo(() => {
```

Comment:
I think `React.HTMLAttributes<HTMLDivElement>['style']` can just be `CssProperties`?